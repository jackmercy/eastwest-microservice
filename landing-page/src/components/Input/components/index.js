export { default as FieldLabel } from './FieldLabel';
export { default as FieldError } from './FieldError';
export { default as FieldInput } from './FieldInput';
