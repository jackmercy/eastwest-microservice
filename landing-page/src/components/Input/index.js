export * from './components';

export { default } from './Input';
export { default as InputNumber } from './InputNumber';
export { default as InputPhone } from './InputPhone';
export { default as InputSearch } from './InputSearch';
export { default as InputPassword } from './InputPassword';
