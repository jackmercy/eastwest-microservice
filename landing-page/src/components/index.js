export { default as SEO } from './SEO';
export { default as Logo } from './Logo';
export { default as Modal } from './Modal';
export { default as Input } from './Input';
export { default as Select } from './Select';
export { default as Button } from './Button';
export { default as Loading } from './Loading';
export { default as Carousel } from './Carousel';
export { default as Textarea } from './Textarea';
export { default as CheckBox } from './CheckBox';
export { default as PopupMenu } from './PopupMenu';
export { default as StaticHTML } from './StaticHTML';

export * from './Input';
export * from './Modal';
