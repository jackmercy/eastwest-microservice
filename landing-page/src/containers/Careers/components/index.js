export { default as SectionTitle } from './SectionTitle';
export { default as MenuScrollableLayout } from './MenuScrollableLayout';
