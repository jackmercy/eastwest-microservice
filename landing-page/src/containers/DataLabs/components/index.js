export { default as Teaser } from './Teaser';
export { default as JoinUs } from './JoinUs';
export { default as Become } from './Become';
export { default as HowItWorks } from './HowItWorks';
export { default as ReadyToStart } from './ReadyToStart';
