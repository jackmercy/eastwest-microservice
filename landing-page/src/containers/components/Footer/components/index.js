export { default as Register } from './Register';
export { default as CopyRight } from './CopyRight';
export { default as SummaryInfo } from './SummaryInfo';
export { default as SectionTitle } from './SectionTitle';
export { default as SwitchLanguages } from './SwitchLanguages';
