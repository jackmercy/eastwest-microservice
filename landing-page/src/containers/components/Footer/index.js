import dynamic from 'next/dynamic';

export default dynamic(() => import('./Footer'), { ssr: false });
