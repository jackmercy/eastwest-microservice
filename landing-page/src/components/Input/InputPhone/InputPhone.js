import React, { memo, useCallback } from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import { FieldLabel, FieldInput, FieldError } from '../components';

import useStyles from './inputPhone.styles';

const InputPhone = ({
  name,
  label,
  error,
  dataTest,
  required,
  onChange,
  className,
  onKeyDown,
  inputClassName,
  labelClassName,
  errorClassName,
  ...restProps
}) => {
  const classes = useStyles();
  const formatPhone = useCallback((value) => {
    if (!value) return value;
    let formatted = String(value);
    // Remove leading 0: 00123 => 0123
    formatted = formatted.replace(/^(-)?0+(0\.|\d)/, '0$1$2');
    // Remove more than one plus except first
    if (formatted.split('')[0] === '+') {
      formatted = `+${formatted.replace(/[+]/g, '')}`;
    } else {
      // Remove plus not first
      formatted = formatted.replace(/[+]/g, '');
    }
    // Replace mutilple space to single space: 01  2 => 01 2
    formatted = formatted.replace(/\s\s+/g, ' ');

    return formatted;
  }, []);

  const handleKeyDown = (e) => {
    const charCode = e.keyCode || e.which;

    if (
      !(
        // allow delete, backspace, dot, move, +, space
        (
          [8, 16, 32, 37, 38, 39, 40, 46, 107, 110, 187, 190].includes(
            charCode,
          ) ||
          (charCode >= 48 && charCode <= 57) || // allow number
          (charCode >= 96 && charCode <= 105)
        ) // allow number
      )
    ) {
      e.preventDefault();
      return;
    }

    if (onKeyDown) {
      onKeyDown(e);
    }
  };

  return (
    <div
      className={classNames(classes.root, className)}
      data-test={dataTest || 'component-input-phone-root'}
    >
      {label ? (
        <FieldLabel
          label={label}
          htmlFor={name}
          required={required}
          className={labelClassName}
        />
      ) : null}
      <FieldInput
        name={name}
        hasError={!!error}
        required={required}
        className={inputClassName}
        onKeyDown={handleKeyDown}
        autoComplete="off"
        onChange={({ target: { name, value } }) => {
          onChange({
            name,
            value: formatPhone(value),
          });
        }}
        {...restProps}
        type="text"
      />
      {error ? <FieldError error={error} className={errorClassName} /> : null}
    </div>
  );
};

InputPhone.propTypes = {
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  label: PropTypes.node,
  error: PropTypes.string,
  required: PropTypes.bool,
  onKeyDown: PropTypes.func,
  dataTest: PropTypes.string,
  className: PropTypes.string,
  inputClassName: PropTypes.string,
  labelClassName: PropTypes.string,
  errorClassName: PropTypes.string,
};
InputPhone.defaultProps = {
  error: '',
  dataTest: '',
  className: '',
  required: false,
  label: undefined,
  inputClassName: '',
  labelClassName: '',
  errorClassName: '',
  onKeyDown: undefined,
};

export default memo(InputPhone);
