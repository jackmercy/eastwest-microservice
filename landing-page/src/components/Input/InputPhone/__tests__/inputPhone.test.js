import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { FieldLabel, FieldInput, FieldError } from '../../components';
import InputPhone from '../InputPhone';

describe('Testing Input component > InputPhone', () => {
  let wrapper;
  const props = {
    label: 'Phone',
    name: 'phone',
    error: 'some error',
    onChange: jest.fn(),
  };

  beforeAll(() => {
    wrapper = mount(<InputPhone {...props} />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<InputPhone {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a FieldLabel', () => {
    expect(wrapper.find(FieldLabel)).toHaveLength(1);
  });

  it('should have a FieldInput', () => {
    expect(wrapper.find(FieldInput)).toHaveLength(1);
  });

  it('should have a FieldError', () => {
    expect(wrapper.find(FieldError)).toHaveLength(1);
  });
});
