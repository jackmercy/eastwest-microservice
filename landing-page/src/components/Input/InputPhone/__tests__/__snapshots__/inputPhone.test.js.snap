// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Input component > InputPhone should render match snapshot 1`] = `
<div
  className="root-0-2-1"
  data-test="component-input-phone-root"
>
  <Memo(FieldLabel)
    className=""
    htmlFor="phone"
    label="Phone"
    required={false}
  />
  <Memo(FieldInput)
    autoComplete="off"
    className=""
    hasError={true}
    name="phone"
    onChange={[Function]}
    onKeyDown={[Function]}
    required={false}
    type="text"
  />
  <Memo(FieldError)
    className=""
    error="some error"
  />
</div>
`;
